src/spdx_matcher/__init__.py

```diff
@@ -226,123 +227,158 @@ def _fetch_json(url: str, timeout=10) -> dict:
-                        f"license_data {json.dumps(license_data)} time {execution_time}"
-                    )
+                node["matchCost"] = execution_time
+                node["matchConfidence"] = match
+
+            if match < 0.99:
+                print(
+                    f"Unable to match {match} full_match:{full_match}  exmplar "
+                    f"text to regexp for license {json_detail_data['licenseId']}  "
+                    f"license_data {json.dumps(license_data)} time {execution_time}"
+                )
+            else:
+                print(
+                    f"Success to match and full_match:{full_match} exmplar text "
+                    f"to regexp for license {json_detail_data['licenseId']} "
+                    f"license_data {json.dumps(license_data)} time {execution_time}"
+                )
+
+    if "regexpForMatch" in node:
+        node["matchCost"] = node["matchCost"] if "matchCost" in node else 100.0
+        node["text_length"] = len(json_detail_data[matcher_config.text])
+    json_detail_data[matcher_config.name] = node
+
+
+def _process_license(license: dict) -> dict:
+    if "detailsUrl" in license:
+        json_detail_data = _fetch_json(license["detailsUrl"])
+        json_detail_data.pop("licenseTextHtml", None)
+        json_detail_data.pop("standardLicenseHeaderHtml", None)
+
+        _process_license_regexp(json_detail_data, TextMatcher())
+        _process_license_regexp(json_detail_data, HeaderMatcher())
 
         for k in json_detail_data:
             if k not in license:
                 license[k] = json_detail_data[k]
         return {
             license["licenseId"]: {
                 "name": license["name"],
-                "licenseText": license["licenseText"],
-                "regexpForMatch": license["regexpForMatch"],
-                "matchCost": license["matchCost"] if "matchCost" in license else 100.0,
-                "text_length": len(license["licenseText"]),
-                "matchConfidence": license["matchConfidence"],
-                "isDeprecatedLicenseId": license["isDeprecatedLicenseId"],
+                "metadata": {
+                    "licenseText": license["licenseText"],
+                    "standardLicenseHeader": license.get("standardLicenseHeader", None),
+                    "reference": license["reference"],
+                    "seeAlso": license.get("seeAlso", []),
+                    "isDeprecatedLicenseId": license.get("isDeprecatedLicenseId", False),
+                },
+                "header": license[HeaderMatcher.name],
+                "headerRegexpExists": True if "regexpForMatch" in license[HeaderMatcher.name] else False,
+                "text": license[TextMatcher.name],
+                "textRegexpExists": True if "regexpForMatch" in license[TextMatcher.name] else False,
             }
         }
 
 
+def _process_exception_regexp(json_detail_data: dict, matcher_config: BaseMatcher) -> dict:
```

Comment:
All the new functions in this file are  use to reduce the logic to handle both `header` and `test` match at the same time